### PR TITLE
Adds vault analytics

### PR DESCRIPTION
**Pull request by timongll**
- Includes Next Indicative Strike, Hit Ratio and Average Drawdown
image
image

### CODE REVIEW

File: shared/src/hooks/useAirtableVaultAnalytics.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        averageLoss: 0,
+      },
+    ])
+  ) as HitRatioResponses,
+  loading: true,
+};
+
+const airtableValueArray = ["hitRatio", "averageLoss"];
+
+dotenv.config();
```
Comment (cjinghong) [inline shared/src/hooks/useAirtableVaultAnalytics.ts:33]: why is this here? It should be in the `index.ts` file

File: shared/src/hooks/useFetchV2VaultData.ts [hunk 2]
```diff
@@ -65,27 +71,55 @@ const useFetchV2VaultData = (): V2VaultData => {
...
+          | { newStrikePrice: BigNumber; newDelta: BigNumber }
         >[] = [
           contract.totalBalance(),
           contract.cap(),
           contract.pricePerShare(),
           contract.vaultState(),
+          strikeSelectionContract
+            ? strikeSelectionContract.getStrikePrice(
+                expiryTimestamp,
+                vault.includes("-P-")
```
Comment (cjinghong) [inline shared/src/hooks/useFetchV2VaultData.ts:115]: is this checking for Put vaults? could we use `isPutVault`?

File: shared/src/utils/derebitMath.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  return getStrikeAtDeltaFromDeribit(
+    isPut,
+    spot,
+    step,
+    linearlyInterpolateOptionChain(options, step),
+    isPut ? -0.1 : 0.1
+  );
+};
+
+const getStrikeAtDeltaFromDeribit = (
```
Comment (cjinghong) [inline shared/src/utils/derebitMath.ts:54]: could be a good idea to unit test this file `deribitMath.test.ts`

File: webapp/src/hooks/derebitContext.tsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+  useRef,
+  useState,
+} from "react";
+import Queue from "smart-request-balancer";
+import {
+  getStrikePricesForOption,
+  getDeribitInstrumentDetails,
+  Option,
+  getInstrumentName,
+} from "shared/lib/utils/derebit";
```
Comment (cjinghong) [inline webapp/src/hooks/derebitContext.tsx:17]: smoll typo, `derebit` should be `deribit`

**Review by cjinghong — CHANGES_REQUESTED** [review]
added comments

**Review by cjinghong — APPROVED** [review]

**Review by obadj3 — DISMISSED** [review]
lgtm

**Review by obadj3 — APPROVED** [review]
lgtm
